subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerDaemonIntegrationTest.groovy

```diff
@@ -30,6 +30,8 @@ import static org.hamcrest.CoreMatchers.notNullValue
 
 @IntegrationTestTimeout(90)
 class WorkerDaemonIntegrationTest extends AbstractWorkerExecutorIntegrationTest {
+
+    @Requires(TestPrecondition.WORKING_DIR)
     def "sets the working directory to the project directory by default during worker execution"() {
```

Comment:
Should we remove that test and behavior instead, always leaving the working dir in a temp directory? That would make it more consistent than different behavior on JDK 8 and 11.

Reply:
I absolutely agree. We cannot remove it without deprecating it first though, can we?

Reply:
Good question, how would you deprecate it though? Nag users into activating an opt-out flag? I'm not sure the reliance on the working dir is common enough for that.

Reply:
I'll look into it and open a separate issue/PR.